metrics/plugins/graphite/init.lua

```diff
@@ -0,0 +1,79 @@
+local socket = require('socket')
+local fiber = require('fiber')
+local metrics = require('metrics')
+local checks = require('checks')
+local log = require('log')
+
+local graphite = {}
+
+-- Default values
+local DEFAULT_PREFIX = 'tarantool'
+local DEFAULT_HOST = '127.0.0.1'
+local DEFAULT_PORT = 2003
+local DEFAULT_SEND_INTERVAL = 2
+
+-- Constants
+local LABELS_SEP = ';'
+
+local function format_observation(prefix, obs)
+    local metric_path = ('%s.%s'):format(prefix, obs.metric_name)
+
+    if next(obs.label_pairs) then
+        local label_pairs_str_parts = {}
+        for label, value in pairs(obs.label_pairs) do
+            table.insert(label_pairs_str_parts, ('%s=%s'):format(label, value))
+        end
+        local label_pairs_str = table.concat(label_pairs_str_parts, LABELS_SEP)
+        metric_path = metric_path .. LABELS_SEP .. label_pairs_str
+    end
+
+    local ts = tostring(obs.timestamp):sub(1, -4) -- remove ULL suffix
+    local graph = ('%s %s %s\n'):format(metric_path, obs.value, ts)
+
+    return graph
+end
+
+local function graphite_worker(opts)
+    fiber.name('metrics_graphite_worker')
+
+    while true do
+        local observations = metrics.collect()
+        for _, obs in ipairs(observations) do
+            local data = format_observation(opts.prefix, obs)
+            local numbytes = opts.sock:sendto(opts.host, opts.port, data)
+            if numbytes == nil then
+                log.info('Error while sending to host %s port %s data %s',
```

Comment:
This is an error.
Maybe better use`log.error()` then?

Reply:
Yes, you are right